src/main/scala/com/microsoft/azure/cosmosdb/spark/streaming/CosmosDBSource.scala

```diff
@@ -106,7 +107,7 @@ private[spark] class CosmosDBSource(sqlContext: SQLContext,
 
```

Comment:
if (shouldInferSchema && customSchema.isEmpty) ? 

Because, the defaultInferSchema value is true and if missed to be set to false even with the customSchema being passed - it will not take effect.

Reply:
Good catch - thanks! Fixed in next iteration.